### PR TITLE
WIP: Tab Redesign

### PR DESCRIPTION
**Pull request by jannuary**
Ephy uses HdyTabBar now, it would be nice to have styling that mimics it for consistency here.

Pretty dirty and unfinished, and only works with the light theme now.

- [x] Clean things up
- [ ] Dark, HC, Yaru, etc. colors
- [ ] Auto hide close button (properly)
- [x] Lock behind a preference
image

### CODE REVIEW

**Comment by rafaelmardojai** [comment]
Looks good! 

We should put these changes behind a preference like `gnomeTheme.newTabs` before making them default.

**Comment by jannuary** [comment]
Hm, I might need help implementing the close button autohide properly, here's how Web does it:
Peek 2021-02-22 19-28

**Comment by jannuary** [comment]
Can't particularly test alternative color schemes or implement close-button autohide at the moment, so unmarking as WIP and requesting a review.

File: theme/parts/tabsbar.css [hunk 1]
```diff
@@ -316,3 +316,70 @@ tab[selected]:-moz-window-inactive {
...
+	.tab-close-button {
+		border-radius: 99px;
+		opacity: 1;
+	}
+	:root:not(:-moz-window-inactive) .tab-close-button:hover {	
+		background-image: none;
+		background-size: 24px;
+		background-color: var(--gnome-tabbar-close-hover);
+		box-shadow: 0px 0px transparent; 
+	    border: 0px solid transparent;
```
Comment (rafaelmardojai) [inline theme/parts/tabsbar.css:373]: This should be applied to `.tab-close-button` it self or will do a jump on hover.

**Comment by rafaelmardojai** [comment]
Is this what you want?
output

**Comment by kdwk** [comment]
Can't wait to try this out

**Comment by jannuary** [comment]
Marking as a draft because Firefox updated and broke this. Apologies for letting this PR sit around for months.

**Comment by rafaelmardojai** [comment]
@jannuary I'm thinking on integrating this directly to the `beta` branch. I have some drafts locally. But no worry, I will give you the credit on the commit :P

**Comment by jannuary** [comment]
Rebased on the beta branch, fixed couple of things, close button autohide still buggy though:
image

**Comment by rafaelmardojai** [comment]
Nice, I think we can remove the preference and apply the change directly, with those Firefox breakages we don't want to maintain two tabs styles 😅.

Could you also rebase again and merge against `beta`?

**Pull request closed without merging**